storage/src/mmr/iterator.rs

```diff
@@ -109,6 +109,63 @@ pub(crate) fn nodes_needing_parents(peak_iterator: PeakIterator) -> Vec<u64> {
     peaks
 }
 
+/// Returns the position of the oldest provable node in the represented MMR.
+pub(crate) fn oldest_provable_pos(
+    peak_iterator: PeakIterator,
+    oldest_remembered_node_pos: u64,
+) -> u64 {
+    for (peak_pos, height) in peak_iterator {
+        if peak_pos >= oldest_remembered_node_pos {
```

Comment:
nit: I like the idea of continuing on `peak_pos < oldest_remembered_node_pos` to reduce nesting

Reply:
me too, changed.